src/solvers/flattening/boolbv.h

```diff
@@ -31,6 +31,8 @@ class bswap_exprt;
 class concatenation_exprt;
 class extractbit_exprt;
 class extractbits_exprt;
+class floatbv_typecast_exprt;
+class ieee_float_op_exprt;
```

Comment:
Why are these forward declared when other types are not.  I can see cases for forward declaration or inclusion but the mix seems unclear.

Reply:
The algorithm is to count the .cpp files that includes this .h file that include the omitted header file. If that is a high percentage, then use the include, otherwise use the forward declaration.

In this case, a lot of .cpp files (46) include boolbv.h but do not need the full floatbv_expr.h (32).